client/test/src/main/java/org/opensearch/client/RestClientTestCase.java

```diff
@@ -81,7 +81,7 @@ protected static void assertHeaders(
         final Header[] defaultHeaders,
         final Header[] requestHeaders,
         final Header[] actualHeaders,
-        final Set<String> ignoreHeaders
+        final List<String> ignoreHeaders
```

Comment:
What is the driver for these changes? The header names should be unique, right?

Reply:
I have reverted the code.

HttpClient 5.4.x has by default enabled HTTP/1.1 TLS Upgrade in https://github.com/apache/httpcomponents-client/pull/542. In my change, I set protocolUpgradeEnabled=default, and user can reset the value.

Reply:
> I have reverted the code.

:+1: 

> HttpClient 5.4.x has by default enabled HTTP/1.1 TLS Upgrade in https://github.com/apache/httpcomponents-client/pull/542. In my change, I set protocolUpgradeEnabled=default, and user can reset the value.

Posted a comment, I don't see a reason why we should disable it

Reply:
@reta, I've conducted tests and found that OpenSearch doesn't support the upgrade to `tls/1.2`; it only supports `h2c`. If protocolUpgradeEnabled is set to true (which is the default), the request will be blocked by netty.

I'm still in the process of learning about Flux, and there seems significant compatibility issues. Should we first set protocolUpgradeEnabled=false, as this won't affect normal usage, and then continue to investigate the compatibility issues when protocolUpgradeEnabled=true?

Reply:
> @reta, I've conducted tests and found that OpenSearch doesn't support the upgrade to `tls/1.2`; it only supports `h2c`. If protocolUpgradeEnabled=true(default), the request will be blocked by netty, and client can't get any response.

Thanks @kkewwei , we use h2c by default but we do support h2 (HTTP/2 + TLS) with security plugin installed, we need to find out where is a gap, the transport shouldn't block. I will try to find time this week to help you out.